devicetypes/rooms-beautiful/rooms-beautiful-curtain.src/rooms-beautiful-curtain.groovy

```diff
@@ -128,31 +150,60 @@ def parse(String description) {
 	}
 }
 
+def off() {
+	zigbee.off() +
+    sendEvent(name: "level", value: 0)
+}
+
+def on() {
+	zigbee.on() +
+    sendEvent(name: "level", value: 100)
+}
+
 def close() {
-	zigbee.off()
+	zigbee.off() +
+    sendEvent(name: "level", value: 0)
 }
 
 def open() {
-	zigbee.on()
+	zigbee.on() +
+    sendEvent(name: "level", value: 100)
 }
 
 def pause() {
 	zigbee.command(0x0102, 0x02) +
-	sendEvent(name: "replay", value: "cont")
+	sendEvent(name: "replay", value: "cont") +
+    sendEvent(name: "windowShade", value: "partially open")
 }
 
 def cont() {
 	zigbee.command(0x0102, 0x02) +
 	sendEvent(name: "replay", value: "pause")
 }
 
+def setLevel(value) {
+	def time
+	if(state.updatedDate == null){
+		time = now()
+	}
+	else{
+		time = now() - state.updatedDate
+	}
+	state.updatedDate = now()
+    log.trace ("Time: ${time}")
+        
+    if (time > 1000 ){
+        zigbee.command(0x0102, 0x05, zigbee.convertToHexString(100-value, 2)) +
+        sendEvent(name: "level", value: value) +
```

Comment:
I'm not sure if this works to append the `log.debug` to the return value but I would suggest logging the message first and then returning only the `zigbee.command` and the `sendEvent`.

Reply:
I have now fixed this.